src/platform/Linux/DnssdImpl.cpp

```diff
@@ -328,6 +328,7 @@ CHIP_ERROR MdnsAvahi::Init(DnssdAsyncReturnCallback initCallback, DnssdAsyncRetu
     CHIP_ERROR error = CHIP_NO_ERROR;
     int avahiError   = 0;
 
+    Shutdown();
```

Comment:
Shutdown seems to return an error. Please log if this fails (I assume we do not want to fail init if shutdown fails).

Reply:
Shutdown only ever returns CHIP_NO_ERROR, there is no codepath that returns anything else. Should this change in the future then indeed we could add a log for it then. Even then it would never cause Init to fail.

Reply:
Shutdown returns CHIP_ERROR. you should handle it.

Reply:
Added log of Shutdown failure as requested.